ingestion/src/metadata/ingestion/source/pipeline/dagster.py

```diff
@@ -100,62 +100,106 @@ def session(self) -> Session:
         return self._session
 
     def get_run_list(self):
+        host_port = self.service_connection.hostPort
+        char_to_replace = {"https://": "", "http://": ""}
+        for key, value in char_to_replace.items():
+            host_port = host_port.replace(key, value)
+
+        host, port = host_port.split(":")
+        try:
+            client = DagsterGraphQLClient(hostname=host, port_number=int(port))
+            result = client._execute(
+                """
+            query AssetNodeQuery {
+            assetNodes {
+                __typename
+                ... on AssetNode {
+                    id
+                    jobNames
+                    groupName
+                    graphName
+                    opName
+                    opNames
+                    jobs{
+                        id
+                        name
+                        description
+                        runs{
+                            id
+                            runId
+                            status
+                            stats{
+                                    ... on RunStatsSnapshot {
+                                            startTime
+                                            endTime
+                                            stepsFailed
+                                        }
+                                }
+                    
+                        }
+                    }
+                
+                
+                    }
+                }
+            }
+        """
+            )
+        except ConnectionError:
+            return False
```

Comment:
This exception handling can be improved because even though you are returning False it will eventually break on like `220` where you are calling this function and running loop on returned value. let's add some error and debug logs here and break the execution in case of connection error